### PR TITLE
FLUID-6355: Pulling in Font Awesome free glyphs for use in custom icon fonts

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-6355

Also updated the linting and other dependencies.

### CODE REVIEW

**Comment by jobara** [comment]
@jhung could you please review this PR. I'm curious as well to know what you think about the structure of the SVG directory. Maybe the fontawesome files should be put somewhere else, or our own SVGs also be in a sub directory.

**Comment by jhung** [comment]
> @jhung could you please review this PR. I'm curious as well to know what you think about the structure of the SVG directory. Maybe the fontawesome files should be put somewhere else, or our own SVGs also be in a sub directory.

I think having our own icons in a sub-directory makes sense. Perhaps `svg/infusion`?

**Comment by jobara** [comment]
@jhung, I've moved our fist party icons into an infusion directory under "svg".

**Comment by jobara** [comment]
@jhung I've made use of the font awesome icons for the overview panel. You can see it in my other PR, https://github.com/fluid-project/infusion/pull/944. Please let me know what you think of that. Also we are missing the astrisk icon from the infusion-icons, so that one is still using the old icon font.

**Comment by jhung** [comment]
@jobara Here's the Fluid Star SVG
[fluid-star.svg.zip](https://github.com/fluid-project/infusion-icons/files/2611205/fluid-star.svg.zip)

**Comment by jobara** [comment]
@jhung was the Fluid Star added to the infusion icons ai file in ownCloud. Also was it built to the spec in the [visual style guide](https://wiki.fluidproject.org/display/fluid/Infusion+Icons+Visual+Style+Guide).

**Comment by jobara** [comment]
@jhung I've added the new build options, and namespace the infusion icons with the "infusion-" prefix.  Ready for more review.

File: README.md [hunk 1]
```diff
@@ -20,31 +28,48 @@ The following dependencies must be installed first:
...
 
-To compile the font icon simply run `grunt` from the project root.
+To compile all icons into a single font simply run `grunt` from the project root.
 
 ```bash
 grunt
 ```
 
-This will create the icon fonts in the `build` directory. Additionally CSS and HTML files will be generated alongside the fonts. The CSS file can be used for adding the icons to your web page; however, you can manually write your own CSS if preferred. The HTML file can be used to verify the output of the font generation.
+To compile only the infusion icons run the following:
```
Comment (jhung) [inline README.md]: Capitalize "Infusion".

File: README.md [hunk 2]
```diff
@@ -20,31 +28,48 @@ The following dependencies must be installed first:
...
+
+```bash
+grunt build:infusion
+```
+
+To compile only the font awesome icons run the following:
```
Comment (jhung) [inline README.md]: Capitalize "Font Awesome".

**Comment by jobara** [comment]
@jhung I've addressed your comments. Ready for more review.

**Pull request merged**